src/Middleware/HttpLogging/test/HttpLoggingMiddlewareTests.cs

```diff
@@ -0,0 +1,757 @@
+
+            Assert.Contains(TestSink.Writes, w => w.Message.Contains(expected));
+        }
+
+        [Theory]
+        [InlineData("text/plain")]
+        [InlineData("text/html")]
+        [InlineData("application/json")]
+        [InlineData("application/xml")]
+        [InlineData("application/entity+json")]
+        [InlineData("application/entity+xml")]
+        public async Task VerifyDefaultMediaTypeHeaders(string contentType)
+        {
+            // media headers that should work.
+            var expected = new string('a', 1000);
+            var options = CreateOptionsAccessor();
+            options.CurrentValue.LoggingFields = HttpLoggingFields.RequestBody;
+
+            var middleware = new HttpLoggingMiddleware(
+                async c =>
+                {
+                    var arr = new byte[4096];
+                    while (true)
+                    {
+                        var res = await c.Request.Body.ReadAsync(arr);
+                        if (res == 0)
+                        {
+                            break;
+                        }
+                    }
+                },
+                options,
+                LoggerFactory);
+
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.ContentType = contentType;
+            httpContext.Request.Body = new MemoryStream(Encoding.UTF8.GetBytes(expected));
+
+            await middleware.Invoke(httpContext);
+
+            Assert.Contains(TestSink.Writes, w => w.Message.Contains(expected));
+        }
+
+        [Theory]
+        [InlineData("application/invalid")]
+        [InlineData("multipart/form-data")]
+        public async Task RejectedContentTypes(string contentType)
+        {
+            // media headers that should work.
+            var expected = new string('a', 1000);
+            var options = CreateOptionsAccessor();
+            options.CurrentValue.LoggingFields = HttpLoggingFields.RequestBody;
+
+            var middleware = new HttpLoggingMiddleware(
+                async c =>
+                {
+                    var arr = new byte[4096];
+                    var count = 0;
+
+                    while (true)
+                    {
+                        var res = await c.Request.Body.ReadAsync(arr);
+                        if (res == 0)
+                        {
+                            break;
+                        }
+                        count += res;
+                    }
+
+                    Assert.Equal(1000, count);
+                },
+                options,
+                LoggerFactory);
+
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.ContentType = contentType;
+            httpContext.Request.Body = new MemoryStream(Encoding.UTF8.GetBytes(expected));
+
+            await middleware.Invoke(httpContext);
+
+            Assert.DoesNotContain(TestSink.Writes, w => w.Message.Contains(expected));
```

Comment:
Should we have a log for this scenario? `"RequestBody: <Unsupported media type>"`

Reply:
Not sure if that's how we should log that?

Reply:
Was it discussed in API review?

Reply:
I don't think so.

Reply:
You added a log right? Add an assert?